Client/NoOpRunner.Client/Rendering/GameWindowRenderer.cs

```diff
@@ -1,48 +1,188 @@
+                    }
 
-                canvas.Children.Add(rec);
+                    if (player.StateMachine.IsTurning)
+                    {
+                        canvas.Children[i].SetValue(UIElement.RenderTransformProperty,
+                            player.StateMachine.IsTurnedLeft
+                                ? new ScaleTransform() {ScaleX = -1}
+                                : new ScaleTransform() {ScaleX = 1});
+                    }
+
+                    #region Animation rectangle move, need non static method
+
+                    SetAnimation(canvas.Children[i], rectangleWidth * playerPixels[i].X,
+                        rectangleHeight * playerPixels[i].Y);
+
+                    #endregion
+                }
+            }
+            else
+            {
+                canvas.Children.Clear();
+
+                foreach (var windowPixel in playerPixels) 
+                {
+                    var playerPixel = new GifImage()
+                    {
+                        Width = rectangleWidth, Height = rectangleHeight,
+                        GifSource = player.StateMachine.GetStatusUri()
+                    };
+
+                    Canvas.SetLeft(playerPixel, rectangleWidth * windowPixel.X);
+                    Canvas.SetBottom(playerPixel, rectangleHeight * windowPixel.Y);
+
+                    canvas.Children.Add(playerPixel);
+                    
+                    playerPixel.StartAnimation();
+                }
             }
         }
+        private static void SetAnimation(UIElement uiElement, double newPosX, double newPosY)
+        {
+            var lastPositionY = (double) uiElement.GetValue(Canvas.BottomProperty);
+            var lastPositionX = (double) uiElement.GetValue(Canvas.LeftProperty);
+
+            var moveAnimX =
+                new DoubleAnimation(lastPositionX, newPosX, TimeSpan.FromMilliseconds(1000/20));
+            var moveAnimY =
+                new DoubleAnimation(lastPositionY, newPosY, TimeSpan.FromMilliseconds(1000/20));
 
-        private static Dictionary<Core.Enums.Color, SolidColorBrush> ColorBrushMap = new Dictionary<Core.Enums.Color, SolidColorBrush>
+            uiElement.BeginAnimation(Canvas.LeftProperty, moveAnimX);
+            uiElement.BeginAnimation(Canvas.BottomProperty, moveAnimY);
+        }
+        private static ImageBrush InitImageBrush(Uri uri)
         {
-            { Core.Enums.Color.Red, Brushes.Red },
-            { Core.Enums.Color.Green, Brushes.Green },
-            { Core.Enums.Color.Yellow, Brushes.Yellow },
-            { Core.Enums.Color.Blue, Brushes.Blue },
-            { Core.Enums.Color.Black, Brushes.Black }
-        };
+            var bitmapImage = new BitmapImage();
+            bitmapImage.BeginInit();
+            bitmapImage.UriSource = uri;
+            bitmapImage.EndInit();
+
+            return new ImageBrush(bitmapImage);
```

Comment:
Can't you just pass the Uri to the BitmapImage ctor?

Reply:
Gone